### PR TITLE
Qrcode receiver modal

### PR DESCRIPTION
**Pull request by portdeveloper**
## Description

This PR removes the QR code SVG from the dropdown list, and adds a button to open a modal. See #440 for more details.

https://github.com/scaffold-eth/scaffold-eth-2/assets/108868128/99af222d-08d8-48f7-87ba-5668951d833d

I had to remove the z-index from the Footer.tsx as I could only make it work like that. Open to any other suggestions.

## Additional Information

- [ ] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [ ] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #237 

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address: portdev.eth

### CODE REVIEW

**Comment by austintgriffith** [comment]
this is great! 🔥🫡🔥 thanks sers!

**Comment by austintgriffith** [comment]
(I like the version with the modal the best!)

File: packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx
Comment (technophile-04) [inline packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx:145]: NITPICK : 

Maybe give `className="rounded-xl"` to `QRCodeSVG` ? So that it matches Modal & our kind of general rounded theme 

Screenshot 2023-07-25 at 9 43 56 PM

PS : I think we can also remove the parent's div `space-y-3`

**Comment by technophile-04** [comment]
Tysm @portdeveloper looking great !! 🙌

Even I like this more as compared to #440 and this one will be more visible too while presenting

File: packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx
Comment (portdeveloper) [reply packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx:145]: Looks a lot better, thanks!

**Comment by Pabl0cks** [comment]
I like this one better than #440, more compact and clean. I feel both use cases "Copy address" and "View QR code" work better with this design. Great job @portdeveloper !! 🙌

File: packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx
Comment (rin-st) [inline packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx:88]: I can't see where menu starts. What about changing bg to `bg-secondary`?
before:
Снимок экрана 2023-07-26 в 12 05 16
Снимок экрана 2023-07-26 в 12 10 02


after:
Снимок экрана 2023-07-26 в 12 04 32
Снимок экрана 2023-07-26 в 12 04 39

Also, I don't like gray hover and ability to click when address already copied. But it's daisyui bug and minor thing

**Comment by rin-st** [comment]
Looks good!

Noticed a bug - when I click to the address on modal, blockexplorer is opened in background. But I expect that the address just be copied

https://github.com/scaffold-eth/scaffold-eth-2/assets/25638585/390a7e5b-1aea-4698-900b-f210ab666533

File: packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx
Comment (carletex) [reply packages/nextjs/components/scaffold-eth/RainbowKitCustomConnectButton.tsx:88]: Yes! I noticed the same (but it feels like a more general issue in SE-2), so I created this issue:

https://github.com/scaffold-eth/scaffold-eth-2/issues/441

Not sure if a different bg, a border, or more shadow... I'll ask Andrea too!

**Comment by carletex** [comment]
> Noticed a bug - when I click to the address on modal, blockexplorer is opened in background. But I expect that the address just be copied

Yep, not sure about this.

The <Address> component links to the chain blockexplorer for the displayed address. In localhost, it links to the local blockexplorer, but in live chains, it goes to the corresponding blockexplorer (which can be handy).

Options:
1. Leave it as it's
2. Using `disableAddressLink` in the Address component for that instance. (although it will also affect live chains)
3. Same as 2, but only for local networks.

**Comment by rin-st** [comment]
I forgot that `<Address />` works that way :). Probably we need to discuss it in another issue, but I believe that
- click on address should copy it by default or probably do nothing, since we need to copy much more often than viewing on block explorer (or is it only for me?) and it's not obvious that it links to blockexplorer. 
- Small `View on block explorer` link below the address or some icon after "Copy", for example something like this
image

> 1. Leave it as it's
> 2. Using disableAddressLink in the Address component for that instance. (although it will also affect live chains)
> 3. Same as 2, but only for local networks.

So I think point 2 is good. But open modal -> go to blockexplorer is convenient too, so additional blockexplorer link/icon can help :). Or we can add "View on blockexplorer" option in main menu, so we'll not need it in modal.

**Review by rin-st — APPROVED** [review]

**Comment by rin-st** [comment]
lgtm 👍

**Comment by carletex** [comment]
Let's wait until #455 is merged. We should add the new shadows and the new menu item sizes here too

**Comment by rin-st** [comment]
Looks great! Noticed one micro bug - when you click on copy address, it blinks when content of button changes. First one - click. Second - not.

https://github.com/scaffold-eth/scaffold-eth-2/assets/25638585/f6cbe9c0-a7f4-4986-a45d-12db3ced7e00

**Comment by carletex** [comment]
Good eye @rin-st !! haha

It's the :active pseudo-class when using daisy's `menu` class. In some places where using some daisy SE-2 where we should probably benefit using tw directly (if not, you have to override, sometime with `!`)

Anyway, let's merge this, and maybe we can rethink using the menu / menu-item classes here in another PR

image

**Comment by carletex** [comment]
Great job @portdeveloper ;)

And thank you all for the review!!

**Comment by portdeveloper** [comment]
Thanks! I am honored :)

**Pull request merged**
